### PR TITLE
Refactor metrics to upgrade to v. 3.0.0 

### PR DESCRIPTION
**Pull request by divchenko**
Refactor metrics to upgrade to v. 3.0.0 to use configurable alpha smoothing factor

### CODE REVIEW

File: sensei-core/src/main/java/com/senseidb/indexing/activity/PurgeUnusedActivitiesJob.java [hunk 1]
```diff
@@ -43,19 +47,15 @@
...
 
 public class PurgeUnusedActivitiesJob implements Runnable, PurgeUnusedActivitiesJobMBean {
   private final static Logger logger = Logger.getLogger(PurgeUnusedActivitiesJob.class);
   
   private final CompositeActivityValues compositeActivityValues;
   private final Set<IndexReaderFactory<ZoieIndexReader<BoboIndexReader>>> zoieSystems;
-  private static Timer timer = Metrics.newTimer(new MetricName(PurgeUnusedActivitiesJob.class, "purgeUnusedActivityIndexes"), TimeUnit.MILLISECONDS, TimeUnit.SECONDS);
-  private static Counter foundActivitiesToPurge = Metrics.newCounter(new MetricName(PurgeUnusedActivitiesJob.class, "foundActivitiesToPurge"));
-  private static Counter recentUidsSavedFromPurge = Metrics.newCounter(new MetricName(PurgeUnusedActivitiesJob.class, "recentUidsSavedFromPurge"));
+  private static Timer timer = MetricFactory.newTimer(new MetricName(PurgeUnusedActivitiesJob.class, "purgeUnusedActivityIndexes"));
```
Comment (santip) [inline sensei-core/src/main/java/com/senseidb/indexing/activity/PurgeUnusedActivitiesJob.java:56]: is seconds a default unit here? why did you remove it?

File: sensei-core/src/main/java/com/senseidb/metrics/MetricFactory.java [hunk 2]
```diff
@@ -57,75 +55,66 @@ public static void start() {
...
 
-  /**
-   * @see MetricsRegistry#newTimer(com.yammer.metrics.core.MetricName, java.util.concurrent.TimeUnit, java.util.concurrent.TimeUnit)
-   */
-  public static Timer newTimer(MetricName metricName,
-                        TimeUnit durationUnit,
-                        TimeUnit rateUnit) {
-    return getRegistry().newTimer(metricName, durationUnit, rateUnit);
+  public static Timer newTimer(MetricName metricName) {
+    Timer timer = new Timer(new ExponentiallyDecayingReservoir(DEFAULT_SIZE, DEFAULT_ALPHA));
```
Comment (santip) [inline sensei-core/src/main/java/com/senseidb/metrics/MetricFactory.java:63]: any chance we can make these configurable? why 1028 for size?

File: sensei-core/src/main/java/com/senseidb/indexing/activity/PurgeUnusedActivitiesJob.java (hunk 1 above)
Comment (divchenko) [inline sensei-core/src/main/java/com/senseidb/indexing/activity/PurgeUnusedActivitiesJob.java:56]: yeah, you can't even pass it to new metrics any more

**Comment by santip** [comment]
in general looks good

File: sensei-core/src/main/java/com/senseidb/metrics/MetricFactory.java (hunk 2 above)
Comment (divchenko) [inline sensei-core/src/main/java/com/senseidb/metrics/MetricFactory.java:63]: 1028 is the default value in metrics. It's not going to be a problem. alpha on another hand can be. The best we can do is to mark it as volatile and allow setting from SenseiServer just before everything else is started (unfortunately metrics are used via singleton)
